src/components/DatePickerRange.react.js

```diff
@@ -71,6 +71,16 @@ export default class DatePickerRange extends Component {
                 });
             }
         }
+
+        if (
+            clearable &&
+            !start_date &&
+            !end_date &&
+            (oldMomentDates.start_date !== start_date ||
+                oldMomentDates.end_date !== end_date)
```

Comment:
@Marc-Andre-Rivet  when you click the `x`, both start and end will be set to null, the condition can be either one date changed .